netbox_onboarding/worker.py

```diff
@@ -29,43 +32,59 @@ def onboard_device(task_id, credentials):
     """Process a single OnboardingTask instance."""
     username = credentials.username
     password = credentials.password
+    secret = credentials.secret
 
-    try:
-        ot = OnboardingTask.objects.get(id=task_id)
-    except OnboardingTask.DoesNotExist:
-        # TODO: maybe we started before the DB was done writing it, or maybe it was deleted out from under us?
-        time.sleep(1)
-        ot = OnboardingTask.objects.get(id=task_id)
+    ot = OnboardingTask.objects.get(id=task_id)
 
-    logging.info("START: onboard device")
+    logger.info("START: onboard device")
+    onboarded_device = None
+    try:
+        if ot.ip_address:
+            onboarded_device = Device.objects.get(primary_ip4__address__net_host=ot.ip_address)
+    except Device.DoesNotExist as exc:
+        logger.info(f"Getting device with IP lookup failed: {str(exc)}")
+    except Device.MultipleObjectsReturned as exc:
+        logger.info(f"Getting device with IP lookup failed: {str(exc)}")
+    except ValueError as exc:
+        logger.info(f"Getting device with IP lookup failed: {str(exc)}")
+    except ValidationError as exc:
+        logger.info(f"Getting device with IP lookup failed: {str(exc)}")
```

Comment:
Per our discussion this morning, make sure the logic for each of these cases is correct - some of them should probably lead to an abort rather than a log message and continuation.

Reply:
I improved that, although Value and Validation Errors does not provide too much value here at the moment